tests/test_aging.c

```diff
@@ -0,0 +1,567 @@
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <string.h>
+#include <dirent.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <signal.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <setjmp.h>
+#include <stdarg.h>
+#include <cmocka.h>
+
+#include <libyang/libyang.h>
+
+#include "sysrepo.h"
+#include "sysrepo_types.h"
+#include "test_common.h"
+#include "config.h"
+
+#define TIME_MAX 100000
+#define NUM_OF_FILES 30
+
+typedef struct {
+    sr_conn_ctx_t *conn;
+    sr_session_ctx_t *sess;
+    char *n_path;
+    char *a_path;
+    char *path_to_file;
+    char *pidfile;
+    uint32_t len_npath;
+    uint32_t len_apath;
+    pid_t pid;
+} test_data_t;
+
+/** Allocs memory, memory needs to be freed **/
+char *
+get_test_path(char *which)
+{
+	char *path;
+	if (asprintf(&path, "%s/data/%s", sr_get_repo_path(), which) == -1) {
+		path = NULL;
+	}
+	return path;
+}
+
+int
+create_dir(char *path)
+{
+    char *p, *dup;
+
+    dup = strdup(path);
+    for (p = strchr(dup + 1, '/'); p; p = strchr(p + 1, '/')) {
+        *p = '\0';
+        if (mkdir(dup, 0700) == -1) {
+            if (errno != EEXIST) {
+                *p = '/';
+                free(dup);
+                return -1;
+            }
+        }
+        *p = '/';
+    }
+    free(dup);
+
+    if (mkdir(path, 0700) == -1) {
+        if (errno != EEXIST) {
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+int
+remove_dir(char *path, int *test)
```

Comment:
My advice is to rely on CMake's fixtures for setup and teardown. That way, this thing becomes a simple `rm -r` instead of 50 lines of C code.

The API is confusing -- it doesn't mention the fact that it's trying to work recursively, and the meaning of the `test` argument is not explained at all.